marlowe-dashboard-client/src/Dashboard/State.purs

```diff
@@ -154,71 +158,84 @@ handleAction input@{ currentSlot } UpdateFromStorage = do
 -- the wallet is interested in, compares it to the existing contracts, and either creates new follower
 -- apps or activates pending ones for any contracts that aren't yet being followed
 handleAction input (UpdateFollowerApps companionAppState) = do
-  walletDetails <- use _walletDetails
-  existingContracts <- use _contracts
-  let
-    contractExists marloweParams = elem (Just marloweParams) (view _mMarloweParams <$> values existingContracts)
+  walletCompanionStatus <- use _walletCompanionStatus
+  case walletCompanionStatus of
+    LoadingNewContracts _ -> do
+      -- In the unlikely event that this happens, it means we're still processing the first
+      -- wallet companion state update. To avoid weird things happening, we should let that update
+      -- finish before handling this one. So basically we just try again later.
+      liftAff $ delay $ Milliseconds 100.0
+      handleAction input $ UpdateFollowerApps companionAppState
```

Comment:
Is there a chance we get stuck in an infinite loop here if the load fails?

Reply:
yes, this seems worrying.
In the case that this branch is executed, what flow would unlock this infinite loop? Could we add that as part of the comment?

Reply:
That's a fair point. I'll give it some more thought tomorrow...

Reply:
Hmm, okay, having slept on it, here are my thoughts:

My first thought was: To steer clear of the possibility of getting stuck in a loop, but still be sure that we handle all status update notifications - and handle them all _separately_ without any overlapping - we we would probably need to keep a list of pending notifications, and have a subscription that periodically goes through the list and clears it.

But then I had a second thought: This PR forced me to consider this possibility in the case of the CompanionApp, at the point when we're connecting the wallet, but it's actually a much more general problem. In principle, we could get two status updates for the same Plutus app in quick succession, triggering the handler for the second before we've finished updating the state based on the first. And potentially weird things would then happen.

But third thought: The chances of this happening anywhere are ridiculously small. So although it wouldn't hurt to guard against it, it also doesn't seem like an urgent problem.

So I propose to:

(a) create a (non-urgent) Jira ticket for thinking about this problem a bit more, and perhaps changing how we deal with status updates in general so as to guard against it; and
(b) modify this PR so that it just doesn't attempt to do anything about the problem in this particular case (and so doesn't introduce the problem of potentially getting stuck in a loop).

What do you think?

Reply:
I agree... It would be nice if for `b` we use a Console.log to know if that might affect the user